src/kernel/chainparams.cpp

```diff
@@ -689,15 +689,15 @@ std::optional<ChainType> GetNetworkForMagic(const MessageStartChars& message)
     const auto regtest_msg = CChainParams::RegTest({})->MessageStart();
     const auto signet_msg = CChainParams::SigNet({})->MessageStart();
 
-    if (std::equal(message.begin(), message.end(), mainnet_msg.data())) {
+    if (std::ranges::equal(message, mainnet_msg)) {
```

Comment:
nit: I think we can use `std::array::operator==` here? (edit: or a switch statement would probably be even better)

```suggestion
    if (message == mainnet_msg) {
```

<details>
<summary>git diff on fad0cf6f26</summary>

```diff
diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index 7d06d24be5..a760926ec8 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -689,15 +689,15 @@ std::optional<ChainType> GetNetworkForMagic(const MessageStartChars& message)
     const auto regtest_msg = CChainParams::RegTest({})->MessageStart();
     const auto signet_msg = CChainParams::SigNet({})->MessageStart();
 
-    if (std::ranges::equal(message, mainnet_msg)) {
+    if (message == mainnet_msg) {
         return ChainType::MAIN;
-    } else if (std::ranges::equal(message, testnet_msg)) {
+    } else if (message == testnet_msg) {
         return ChainType::TESTNET;
-    } else if (std::ranges::equal(message, testnet4_msg)) {
+    } else if (message == testnet4_msg) {
         return ChainType::TESTNET4;
-    } else if (std::ranges::equal(message, regtest_msg)) {
+    } else if (message == regtest_msg) {
         return ChainType::REGTEST;
-    } else if (std::ranges::equal(message, signet_msg)) {
+    } else if (message == signet_msg) {
         return ChainType::SIGNET;
     }
     return std::nullopt;

```
</details>

Reply:
That makes sense, but I don't quite see the case for changing `GetNetworkForMagic` to accept a span in the future, so it feels a bit like premature optimization to me, but I may be missing context.

It's a nit, and the commit is orthogonal to the PR anyway, so this is the last I'll comment on it - up to you what you prefer, but I feel like this would probably be the most elegant implementation:

```cpp
std::optional<ChainType> GetNetworkForMagic(const MessageStartChars& message)
{
    static const std::array<std::pair<MessageStartChars, ChainType>, 5> network_map = {{
        {CChainParams::Main()->MessageStart(), ChainType::MAIN},
        {CChainParams::TestNet()->MessageStart(), ChainType::TESTNET},
        {CChainParams::TestNet4()->MessageStart(), ChainType::TESTNET4},
        {CChainParams::RegTest({})->MessageStart(), ChainType::REGTEST},
        {CChainParams::SigNet({})->MessageStart(), ChainType::SIGNET}
    }};

    auto it = std::ranges::find_if(network_map, [&message](const auto& pair) { return pair.first == message; });
    return (it != network_map.end()) ? std::make_optional(it->second) : std::nullopt;
}
```